src/librustdoc/html/render.rs

```diff
@@ -1743,7 +1743,16 @@ fn item_module(w: &mut fmt::Formatter, cx: &Context,
             (Some(stability::Stable), Some(stability::Unstable)) => return Ordering::Less,
             _ => {}
         }
-        i1.name.cmp(&i2.name)
+        name_key(&i1.name).cmp(name_key(&i2.name))
+    }
+
+    fn name_key(name: &str) -> (&str, u64) {
+        let split = s.bytes().rposition(|b| b < b'0' || b'9' < b).unwrap_or(0);
+        // get rid of leading zeroes
+        let split = split + s[split..].bytes().position(|b| b != b'0').unwrap_or(0);
+        let strkey = &s[..split];
+        let intkey = s[split..].parse().unwrap_or(0);
```

Comment:
Actually, you're right, overflowing a u64 can theoretically (but unlikely) happen. Fixed.